my-app/src/app/modules/courses-page/services/courses-handler.service.ts

```diff
@@ -3,45 +3,48 @@ import {Injectable} from '@angular/core';
 import {COURSE_DATA} from "@assets/mocks/course-data.mock";
 
 import {ICourse} from "@shared/";
+import {BehaviorSubject} from "rxjs";
 
 export interface ICoursesHandlerService {
-  getList(): ICourse[];
+  getList(): BehaviorSubject<ICourse[]>;
 
   getItemById(id: number): ICourse | undefined;
 
-  removeItem(id: number): ICourse[];
+  removeItem(id: number): void;
 }
 
 @Injectable({
   providedIn: 'root'
 })
 
 export class CoursesHandlerService implements ICoursesHandlerService {
-  private initialCourses: ICourse[] = COURSE_DATA;
-  private visibleCourses: ICourse[] = structuredClone(this.initialCourses);
+  private existingCourses$: BehaviorSubject<ICourse[]> = new BehaviorSubject(structuredClone(COURSE_DATA));
 
-  constructor() {
+  public getList(): BehaviorSubject<ICourse[]> {
+    return this.existingCourses$;
   }
 
-  public getList(): ICourse[] {
-    return this.visibleCourses;
+  public createCourse(courseData: ICourse): void {
+    let copiedCourses: ICourse[] = structuredClone(this.existingCourses$.getValue());
+
+    copiedCourses.push(courseData);
+
+    this.existingCourses$.next(copiedCourses);
   }
 
-  // public createCourse(properties: Course, arr: Course[]): Course[] {
-  //   arr.push(properties);
-  //   return arr;
-  // }
-  //
   public getItemById(id: number): ICourse | undefined {
-    return this.visibleCourses.find(item => item.id === id);
+    return this.existingCourses$.getValue().find((item: ICourse) => item.id === id);
+  }
+
+  public updateItem(courseData: ICourse): void {
+    const copiedCourses: ICourse[] = structuredClone(this.existingCourses$.getValue());
+
+    this.existingCourses$.next(copiedCourses.map(item => item.id === courseData.id ? courseData : item));
   }
 
-  //
-  // public updateItem() {
-  // }
+  public removeItem(id: number): void {
+    const updatedCourses: ICourse[] = this.existingCourses$.getValue().filter((item: ICourse) => item.id !== id);
```

Comment:
`subject.getValue()` is called many time here. Probably it makes sense to have non-subject regular private array here. It'll be easier to work with it, and then just to pass it into `next()` when we need to notify subscribers about changes.

Reply:
Indeed, thanks a lot for the comments